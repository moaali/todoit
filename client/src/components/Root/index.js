import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { inject, observer } from 'mobx-react'

import { onKeyPairsPress } from 'services';

@inject('store')
@observer
class Root extends Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.arrayOf(PropTypes.node),
    ]).isRequired,
    store: PropTypes.shape({
      sidebarStore: PropTypes.shape({}).isRequired,
    }).isRequired,
  }

  constructor(props) {
    super(props)
    this.sidebarStore = this.props.store.sidebarStore;
  }

  componentDidMount() {
    onKeyPairsPress('b', { ctrlKey: true }, () => {
      this.sidebarStore.toggleSidebarVisibility();
    })
  }

  render() {
    const { children } = this.props;
    const layoutClass = this.sidebarStore.isSidebarShown ? '' : 'collapsed';

    return (
      <div className={layoutClass}>{children}</div>
    );
  }
}

export default Root;
