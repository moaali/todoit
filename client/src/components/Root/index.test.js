import React from 'react'
import { Provider } from 'mobx-react'

import stores from 'stores'
import Root from './'

const RootComponent = () => (
  <Provider store={{...stores}}>
    <Root />
  </Provider>
);

describe('Root', () => {
  it('should match snapshot', () => {
    const rootElement = shallow(<RootComponent />);
    expect(toJson(rootElement)).toMatchSnapshot();
  })
})
