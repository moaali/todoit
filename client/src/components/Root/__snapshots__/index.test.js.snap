// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should match snapshot 1`] = `
<Provider
  store={
    Object {
      "offcanvasStore": OffcanvasStore {},
      "searchStore": SearchStore {},
      "sidebarStore": SidebarStore {},
      "todosStore": TodosStore {
        "activeCategory": "all",
        "applyFilters": [Function],
        "currentOffset": 0,
        "displayedTodos": Array [],
        "filterByCategory": [Function],
        "filterByCompleted": [Function],
        "filterByQuery": [Function],
        "filters": Object {
          "category": "all",
          "completed": null,
          "query": "",
        },
        "findById": [Function],
        "loadChunkSize": 10,
        "loadTodos": [Function],
        "meta": Object {
          "all": 0,
          "completed": 0,
          "pending": 0,
        },
        "requestStatus": "error",
        "shownTodo": null,
        "todos": Array [],
      },
    }
  }
>
  <inject-Root-with-store />
</Provider>
`;
